perfkitbenchmarker/linux_benchmarks/spark_benchmark.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2014 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs a jar using a cluster that supports Apache Spark.
+
+This benchmark takes a jarfile and class name, and runs that class
+using an Apache Spark cluster.  The Apache Spark cluster can be one
+supplied by a cloud provider, such as Google's Dataproc.
+
+It records how long the job takes to run.
+
+For more on Apache Spark, see: http://spark.apache.org/
+"""
+
+import datetime
+import logging
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import flags
+
+
+
+BENCHMARK_NAME = 'spark'
+BENCHMARK_CONFIG = """
+spark:
+  description: Run a jar on a spark cluster.
+  spark_service:
+    spark_service_type: managed
+    num_workers: 4
+"""
+
+flags.DEFINE_string('spark_jarfile', None,
```

Comment:
It would be nice if the benchmark could be run without requiring users to supply their own jar file. Is there anything close to a canonical spark benchmark?